Source/Shared/Paths.cs

```diff
@@ -3,36 +3,43 @@
 
 namespace CodeImp.Bloodmasters;
 
-public static class Paths
+public class Paths
 {
     private const string DevModeMarkerFileName = ".bloodmasters.dev.marker";
     private const string DevSolutionRootMarkerFileName = ".bloodmasters.solution-root.marker";
     private const string TargetFrameworkForExecutables = "net7.0-windows";
-    private const string BuildConfiguration =
+    private const string _buildConfiguration =
 #if DEBUG
         "Debug";
 #else
         "Release";
 #endif
 
+    public Paths(bool isDev = false)
+    {
+        IsDevModeBuild = File.Exists(Path.Combine(AppBaseDir, DevModeMarkerFileName)) || isDev;
+    }
+
+    public static Paths Instance { get; } = new ();
+
     /// <summary>Directory of the entry binary.</summary>
     /// <remarks>In dev mode, it looks like <c>Source/Client/bin/Debug/net7.0-windows</c>.</remarks>
-    private static readonly string AppBaseDir = GetAppBaseDir();
```

Comment:
I propose to make all these members (non-static) readonly fields, and not properties that get re-calculated on every access.